src/app/[locale]/selected-filters.tsx

```diff
@@ -0,0 +1,114 @@
+import {SearchResultFilter} from '@/lib/dataset-fetcher';
+import {Dispatch} from 'react';
+import {useTranslations} from 'next-intl';
+import {XMarkIcon} from '@heroicons/react/24/solid';
+
+interface TagProps {
+  name?: string;
+  remove: () => void;
+}
+
+function Tag({name, remove}: TagProps) {
+  return (
+    <span className="flex items-center" data-test="selectedFilterTag">
+      <span className="m-1 inline-flex items-center rounded-full border border-gray-200 bg-white py-1.5 pl-3 pr-2 text-xs font-medium text-gray-900">
+        <span>{name}</span>
+        <button
+          data-test="removeSelectedFilter"
+          type="button"
+          className="ml-1 inline-flex flex-shrink-0 rounded-full p-1 text-gray-400 hover:bg-gray-200 hover:text-gray-500"
+          onClick={remove}
+        >
+          <XMarkIcon className="h-3 w-3" />
+        </button>
+      </span>
+    </span>
+  );
+}
+
+interface Props {
+  filters: {
+    searchResultFilters: SearchResultFilter[];
+    selectedFilters: string[];
+    setSelectedFilters: Dispatch<string[]>;
+  }[];
+  query: {
+    value: string;
+    setQuery: Dispatch<string>;
+  };
+}
+
+interface RemoveListItem {
+  id: string;
+  selectedFilters: string[];
+  setSelectedFilters: Dispatch<string[]>;
+}
+
+export default function SelectedFilters({filters, query}: Props) {
+  const t = useTranslations('Home');
+
+  function removeListItem({
+    id,
+    selectedFilters,
+    setSelectedFilters,
+  }: RemoveListItem) {
+    setSelectedFilters(selectedFilters.filter(filterId => id !== filterId));
+  }
+  function removeQuery() {
+    query.setQuery('');
+  }
+
+  function clearAllFilters() {
+    removeQuery();
+    filters.forEach(filter => {
+      filter.setSelectedFilters([]);
+    });
+  }
+
+  if (
+    !query.value &&
+    !filters.filter(filter => filter.selectedFilters.length).length
+  ) {
+    return null;
+  }
```

Comment:
Perhaps put this block at the beginning of the function ('fail-fast')? And a comment to make clear what it does?